services/blockchain-app-registry/shared/utils/downloadRepository.js

```diff
@@ -180,48 +197,178 @@ const buildEventPayload = async (allFilesModified) => {
+		const connection = await getDbConnection(MYSQL_ENDPOINT);
+
+		await BluebirdPromise.map(
+			Object.keys(groupedFiles),
+			async (networkName) => {
+				const appsInNetwork = groupedFiles[networkName];
+
+				await BluebirdPromise.map(
+					Object.keys(appsInNetwork),
+					async (appName) => {
+						const appFiles = appsInNetwork[appName];
+						const dbTrx = await startDbTransaction(connection);
+						const filesToBeDeleted = [];
+						const filesToBeMoved = [];
+
+						// Should process app files sequentially as nativetokens.json is dependant on app.json
+						// eslint-disable-next-line no-restricted-syntax
+						for (const modifiedFile of appFiles) {
+							/* eslint-disable no-await-in-loop */
+							const remoteFilePath = modifiedFile.filename;
+							const localFilePath = path.join(appDirPath, remoteFilePath);
+
+							// Delete indexed information if a meta file is updated/deleted
+							if (fileUpdatedStatuses.includes(modifiedFile.status)) {
+								await deleteIndexedMetadataFromFile(localFilePath, dbTrx);
+							}
+
+							// Skip download and indexing for deleted file.
+							// The file should be deleted only after processing of the app is done.
+							// As app.json file is required to delete indexed info of nativetokens.json
+							if (modifiedFile.status === GITHUB_FILE_STATUS.REMOVED) {
+								filesToBeDeleted.push(localFilePath);
+								// eslint-disable-next-line no-continue
+								continue;
+							}
+
+							// Create directory and download latest file
+							const tempFilePath = path.join(tempDownloadDir, remoteFilePath);
+							const dirPath = path.dirname(tempFilePath);
+							await mkdir(dirPath, { recursive: true });
+
+							const result = await getFileDownloadURL(remoteFilePath);
+							await downloadFile(result.data.download_url, tempFilePath);
+							logger.debug(`Successfully downloaded: ${tempFilePath}.`);
+
+							// Update DB with latest metadata file information
+							await indexMetadataFromFile(tempFilePath, dbTrx);
+							logger.debug(`Successfully updated the database with the latest changes of file: ${remoteFilePath}.`);
+
+							// Schedule files to be moved
+							filesToBeMoved.push({
+								from: tempFilePath,
+								to: localFilePath,
+							});
+							/* eslint-enable no-await-in-loop */
+						}
+						await commitDbTransaction(dbTrx);
+
+						// Delete files which are removed from remote
+						await BluebirdPromise.map(
+							filesToBeDeleted,
+							async (filePath) => rm(filePath),
+							{ concurrency: filesToBeDeleted.length },
+						);
+
+						// Move downloaded files
+						await BluebirdPromise.map(
+							filesToBeMoved,
+							async (filePathInfo) => {
+								// Create directory to move file
+								await mkdir(path.dirname(filePathInfo.to));
+								await rename(filePathInfo.from, filePathInfo.to);
+							},
+							{ concurrency: filesToBeMoved.length },
+						);
```

Comment:
> I think, we can directly move (or overwrite existing) files instead of deleting them. Only delete the ones that have the status GITHUB_FILE_STATUS.REMOVED.

Actually we are doing this currently.

> Create an alias mv

Applied.